WalletWasabi/Services/IndexDownloader.cs

```diff
@@ -162,7 +161,10 @@ public void Synchronize(TimeSpan requestInterval)
 
 								continue;
 							}
-							else if (filters == null)
+						}
+						catch (Exception ex)
+						{
+							if (bestKnownFilter != null && ex.Message.Contains($"Provided bestKnownBlockHash is not found: {bestKnownFilter?.BlockHash}."))
```

Comment:
Could you use `catch (HttpRequestException) when (ex.Message != null && ex.Message.StartsWith(""Provided bestKnownBlockHash is not found:"){}`?

Reply:
Done.